### PR TITLE
update dockerfiles to use ubi images and update go to 1.21

### PR DESCRIPTION
**Pull request by HumairAK**

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,92 +1,48 @@
-### Updated base image to golang in order to build with go modules
-### Bazel build cannot work with the Tekton library because the current
```
Comment (DharmitD) [inline backend/Dockerfile:2]: would we need to update this dockerfile too? https://github.com/opendatahub-io/data-science-pipelines-tekton/blob/master/backend/Dockerfile.viewercontroller

Comment (HumairAK) [reply backend/Dockerfile:2]: we don't use viewer controller in odh so no so no

**Review by DharmitD — APPROVED** [review]
/lgtm
/Approve

**Comment by HumairAK** [comment]
```
 KFP Tekton Unit Tests / backend-integration (pull_request) In progress — This check has started... 
```

Is failing because it can't pull the ubi images, we'll fix this in a follow up pr, this is odh only and due to minimal sort for dsp tekton this is not high priority, merging to unblock odh release for dps.

**Pull request merged**